homeassistant/components/comelit/coordinator.py

```diff
@@ -95,9 +95,17 @@ async def _async_update_data(self) -> T:
             await self.api.login()
             return await self._async_update_system_data()
         except (CannotConnect, CannotRetrieveData) as err:
-            raise UpdateFailed(repr(err)) from err
+            raise UpdateFailed(
+                translation_domain=DOMAIN,
+                translation_key="update_failed",
+                translation_placeholders={"error": repr(err)},
+            ) from err
         except CannotAuthenticate as err:
-            raise ConfigEntryAuthFailed from err
+            raise ConfigEntryAuthFailed(
+                translation_domain=DOMAIN,
+                translation_key="cannot_authenticate",
+                translation_placeholders={"error": repr(err)},
```

Comment:
What is the added benefit of the error here?

Reply:
As we need to raise, we need to translate; isn't it ? ;-)